src/main/java/com/connexta/store/poller/TransformStatusPoller.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.store.poller;
+
+import com.connexta.store.config.TransformConfiguration;
+import com.connexta.store.controllers.StoreController;
+import com.connexta.store.rest.models.AddMetadataRequest;
+import com.connexta.store.rest.models.MetadataInfo;
+import com.connexta.store.rest.models.QuarantineRequest;
+import com.connexta.transformation.rest.models.MetadataInformation;
+import com.connexta.transformation.rest.models.Status;
+import com.connexta.transformation.rest.models.TransformationPollResponse;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/** Class capable of polling the transformation poll endpoint. */
+@Slf4j
+public class TransformStatusPoller extends Thread {
+
+  private final WebClient storeWebClient;
+
+  private final WebClient transformWebClient;
+
+  private final BlockingQueue<TransformStatusTask> transformStatusQueue;
+
+  /** Creates a new TransformStatusPoller. */
+  public TransformStatusPoller(
+      BlockingQueue<TransformStatusTask> transformStatusQueue,
+      WebClient transformWebClient,
+      WebClient storeWebClient) {
+    this.transformStatusQueue = transformStatusQueue;
+    this.transformWebClient = transformWebClient;
+    this.storeWebClient = storeWebClient;
+  }
+
+  // TODO 11/20/2019 PeterHuffer: This class should be managed and be able to be properly destroyed
+  @SuppressWarnings("squid:S2189" /* Run forever by design */)
+  @Override
+  public void run() {
+    while (true) {
+      try {
+        doRun();
```

Comment:
❓  Should we add some sort of delay here so we don't flood transform with status requests?

Reply:
We could but I personally don't mind flooding right now. Eventually I think we will have a back-off time and depending on the amount of failures for a status.

Reply:
https://github.com/dyng/polling provides strategy objects for for waiting and for giving up:

These are the strategies for waiting between polling attempts:

- FixedWaitStrategy
- RandomWaitStrategy
- CompositeWaitStrategy
- FibonacciWaitStrategy
- ExponentialWaitStrategy
- IncrementingWaitStrategy

These are the strategies for deciding when to give up:

- NeverStopStrategy
- StopAfterAttemptStrategy
- StopAfterDelayStrategy
- CompositeStopStrategy